src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -498,6 +498,7 @@ var ReactCompositeComponentMixin = {
     var inst = this._instance;
     var childContext = inst.getChildContext && inst.getChildContext();
     if (childContext) {
+      ReactCurrentOwner.processingChildContext = true;
```

Comment:
>Would I be writing my own devtool function or do I use something like ReactDOMInstrumentation.debugTool.onSetValueForProperty(inst, 'processingChildContext', true)
 
I think both. Here’s another example of adding a devtool for warnings: https://github.com/facebook/react/commit/251d6c30b51c6aa9a406f9076c0f895570a710b3

`onSetValueForProperty()` will not be relevant here, it’s only called this way because it is called when React sets a value for DOM property.

In your case, you can add `onBeginProcessingChildContext()` and `onEndProcessingChildContext()`, and a new `ReactInvalidSetStateWarningDevTool` (or something like this) that would set the flag.

Also, since it’s related to `React` and not `ReactDOM` in particular, perhaps it makes sense to use `ReactInstrumentation` introduced in #6068 instead of `ReactDOMInstrumentation`.